src/log.cc

```diff
@@ -139,33 +143,34 @@ save_log (char *uri, void *data)
       return;
     }
 
-  /* XXX - Using gnome_vfs_create with exclusive == true to check for
-           file existence doesn't work with obex.  Why am I not
-           surprised?
-   */
+  file = g_file_new_for_uri (uri);
+  info = g_file_query_info (file, G_FILE_ATTRIBUTE_ACCESS_CAN_WRITE,
+                            G_FILE_QUERY_INFO_NONE, NULL, &error);
 
-  result = gnome_vfs_get_file_info (uri, &info, GNOME_VFS_FILE_INFO_DEFAULT);
-  if (result == GNOME_VFS_OK)
+  if (info)
     {
       ask_custom (dgettext ("hildon-fm", "docm_nc_replace_file"),
 		  dgettext ("hildon-libs", "wdgt_bd_yes"),
 		  dgettext ("hildon-libs", "wdgt_bd_no"),
 		  save_log_cont, uri);
     }
-  else if (result != GNOME_VFS_ERROR_NOT_FOUND)
+  else if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
     {
       /* Remove the previously stored dir if failed */
       if (last_save_log_dir != NULL)
         g_free (last_save_log_dir);
       last_save_log_dir = NULL;
 
       /* Annoy user with error */
```

Comment:
This comment seems excessive now.

Reply:
What do you mean?

Reply:
```
      /* Annoy user with error */
      annoy_user_with_gerror (error, uri, save_log_do_nothing, NULL);
```

This comment doesn't seem like something precious, when you're looking at the line it's dedicated for. Don't you think?

Reply:
Just FYI, I really think nitpicking about comments that I didn't even touch is way out of scope for this pull request.

Reply:
If comments become useless due to changes in the code, this also needs to be fixed, imho.

Reply:
No functionality has changed here.

Reply:
Also let's please stop wasting time on this. I have better things to do.